test/test_model.py

```diff
@@ -36,6 +39,9 @@ def relation_get(self, relation_id, member_name):
             },
         }[relation_id][member_name]
 
+    def relation_set(self, relation_id, key, value):
+        self.relation_set_called = True
```

Comment:
Doesn't smell like a test that will catch potential mistakes we make now or in the future. We could call this with literally anything and our backend would be happy. I suggest saving this data, and double checking that we got the proper data. This can also be taught to raise ValueError instead of us having to externally mock it.

Also, for a separate PR, we need a set of tests for the real backend. For that we can create an executable inside the test itself that will log what it gets, and return an expected response. This will ensure that the interaction between the backend and the external world is sane. Then, we should actually call the backend _through the model_ so that we catch errors in the interaction of the model and the backend. Otherwise our tests will verify only the backend, and the model with a fake backend, but not the interaction between the two.